src/network/networkd-wiphy.c

```diff
@@ -203,3 +238,36 @@ int manager_genl_process_nl80211_wiphy(sd_netlink *genl, sd_netlink_message *mes
 
         return 0;
 }
+
+int manager_udev_process_wiphy(Manager *m, sd_device *device) {
+        sd_device_action_t action;
+        const char *name;
+        Wiphy *w;
+        int r;
+
+        assert(m);
+        assert(device);
+
+        r = sd_device_get_action(device, &action);
+        if (r < 0)
+                return log_device_warning_errno(device, r, "Failed to get udev action, ignoring device: %m");
+
+        r = sd_device_get_sysname(device, &name);
+        if (r < 0)
+                return log_device_warning_errno(device, r, "Failed to get sysname, ignoring device: %m");
+
+        r = wiphy_get_by_name(m, name, &w);
+        if (r < 0)
+                /* This error is not critical, as the corresponding genl message may be received later.
+                 * Let's log in the debug level. */
+                return log_device_debug_errno(device, r,
+                                              "Failed to get Wiphy object, ignoring '%s' uevent: %m",
+                                              device_action_to_string(action));
+
+        if (action == SD_DEVICE_REMOVE)
+                w->dev = sd_device_unref(w->dev);
+        else
+                device_unref_and_replace(w->dev, sd_device_ref(device));
```

Comment:
`device_unref_and_replace(w->dev, action == SD_DEVICE_REMOVE ? NULL : device)` ?

Reply:
Shouldn't this be similar to `manager_udev_process_link()` where we only remove the associated device when nl80211 tells us the device was removed? Or should `manager_udev_process_link()` be made similar to this, so that we also unref the device there?

It'd be great if we could be consistent across link, wiphy and rfkill here, but I'm not super familiar with the details so it might not be possible.